pkg/bundle/base/bundle.go

```diff
@@ -0,0 +1,22 @@
+package base
+
+import (
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/snyk/policy-engine/pkg/data"
+)
+
+type BundleSource string
+
+type Manifest struct {
+	BundleFormatVersion string `json:"bundle_format_version"`
+}
+
+type Bundle interface {
+	Provider() data.Provider
+	BundleFormatVersion() string
+	Manifest() interface{}
+	SourceInfo() SourceInfo
+	Modules() map[string]*ast.Module
+	Document() map[string]interface{}
+	Validate() error
```

Comment:
```suggestion
	Validate() []error
```

I'd prefer returning multiple errors in the interface (even if we just return one right now in the implementation) for the possibility to build a nicer feedback loop.

Reply:
Actually, that makes some other code rather tricky, so let's leave it as `error` for now, we can always introduce something that wraps multiple errors.